src/rasmus_svebestad_ex/ex02/message_entropy.py

```diff
@@ -0,0 +1,34 @@
+# Task B
+
+
+import collections as col
+import math as m
+
+
+def letter_freq(txt):
+    a = txt.lower()
+    letter_count = {x: a.count(x) for x in set(a)}
+    return col.OrderedDict(sorted(letter_count.items(), key=lambda t: t[0]))
+
+
+def entropy(message):
+    n_i = list(letter_freq(message).values())
+
+    p_i = [0 for i in range(len(n_i))]
+    sum_i = 0
+
+    for i in n_i:
+        sum_i += i
+
+    for i in range(len(n_i)):
+        p_i[i] = n_i[i]/sum_i
+
+    s = 0
+    for i in p_i:
+        s += -(i*m.log2(i))
+    return s

```

Comment:
It looks kind of messy in here. Code works but you may want to look at the suggested solution for a better way.